src/base/bittorrent/torrentinfo.cpp

```diff
@@ -76,13 +77,45 @@ TorrentInfo TorrentInfo::load(const QByteArray &data, QString *error) noexcept
 
 TorrentInfo TorrentInfo::loadFromFile(const QString &path, QString *error) noexcept
 {
+    if (error)
+        error->clear();
+
+    QFile file {path};
+    if (!file.open(QIODevice::ReadOnly)) {
+        if (error)
+            *error = QCoreApplication::translate("TorrentInfo", "File open error");
+        return TorrentInfo();
+    }
+
+    const qint64 fileSizeLimit = 100 * 1024 * 1024;  // 100 MB
+    if (file.size() > fileSizeLimit) {
+        if (error)
+            *error = QCoreApplication::translate("TorrentInfo", "File size exceeds max limit %1").arg(fileSizeLimit);
```

Comment:
This won't work as you expected.

Reply:
why not?
`translate()` returns a translated QString and `arg()` replaces the placeholder.

Reply:
Sorry. I just thought it isn't processed by lupdate. But I was wrong.
But isn't Q_DECLARE_TR_FUNCTIONS(TorrentInfo) more convenient?

Reply:
>But isn't Q_DECLARE_TR_FUNCTIONS(TorrentInfo) more convenient?

Sure, done.